Source/santad/SNTDaemonControlController.mm

```diff
@@ -387,8 +389,98 @@ - (void)syncBundleEvent:(SNTStoredEvent *)event relatedEvents:(NSArray<SNTStored
 
 #pragma mark Control Ops
 
-- (void)installSantaApp:(NSString *)appPath reply:(void (^)(BOOL))reply {
-  LOGI(@"Got to the install path: %@", appPath);
+- (BOOL)verifyPathIsSanta:(NSString *)path {
+  if (path.length == 0) {
+    LOGE(@"No path provided");
+    return NO;
+  }
+
+  BOOL isDir;
+  if (![[NSFileManager defaultManager] fileExistsAtPath:path isDirectory:&isDir] || !isDir) {
+    LOGE(@"Installation path is not a directory: %@", path);
+    return NO;
+  }
+
+  NSError *err;
+  MOLCodesignChecker *cc = [[MOLCodesignChecker alloc] initWithBinaryPath:path error:&err];
+
+  if (err) {
+    LOGE(@"Failed to validate install path: %@", err);
+    return NO;
+  }
+
+  if (![cc.teamID isEqualToString:@"ZMCG7MLDV9"] ||
```

Comment:
Instead of hardcoding this, it might be good to get this from the running binary. MOLCodesignChecker has an initWithSelf method you can use for this

Reply:
Spoke offline. While getting the TID dynamically is easy, pairing with the appropriate SID is hard and would require changes to the MOLXPCConnection repo to expose information about the GUI endpoint. Since this would only affect users attempting to rebuild Santa as their own, renamed product, and because that would require several other code changes as well, punting on this effort for now.